sync2/p2p.go

```diff
@@ -0,0 +1,211 @@
+package sync2
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/spacemeshos/go-spacemesh/fetch/peers"
+	"github.com/spacemeshos/go-spacemesh/p2p/server"
+	"github.com/spacemeshos/go-spacemesh/sync2/multipeer"
+	"github.com/spacemeshos/go-spacemesh/sync2/rangesync"
+)
+
+type Dispatcher = rangesync.Dispatcher
```

Comment:
why is the type alias needed?

Reply:
My idea was for `sync2` package to serve as a facade that hides all the implementation details of sync itself beneath it

Reply:
That's fine, I'm still not sure I understand the type-aliasing narrative though. Not sure how the two are related. Unless you're decorating the original type with more functionality I don't see why this should be necessary. It just adds more indirection to an already quite large package. If you need to leak stuff out of the package, maybe better to do it through interfaces instead of type aliasing (that's just my opinion though).

Reply:
Ok it was a remnant from an older iteration where `Dispatcher` did not use the constructor, etc.
Given that `rangesync.OrderedSet` etc. is needed anyway, I removed this type alias.